dimod/constrained.py

```diff
@@ -657,10 +723,19 @@ def substitute_self_loops(self) -> Dict[Variable, Variable]:
         Acts on the objective and constraints in-place.
 
         Returns:
-            A mapping from the integer variable labels to their introduced
+            Mapping from the integer variable labels to their introduced
             counterparts. The constraint enforcing :math:`j == i` uses
             the same label.
 
+        Examples:
+            >>> from dimod import Integer, ConstrainedQuadraticModel
+            >>> i = Integer('i')
+            >>> cqm = ConstrainedQuadraticModel()
+            >>> cqm.add_constraint(i*i <=3, label="i squared")   # doctest: +IGNORE_RESULT
```

Comment:
Likewise, I think showing how to use the mapping would be useful here.

Reply:
Shown, just the line has a `# doctest: +IGNORE_RESULT` because the created variable label is random